src/mongoose_rabbit_worker.erl

```diff
@@ -37,28 +37,20 @@
 -type state() :: #{amqp_client_opts := mongoose_amqp:network_params(),
                    connection := pid(),
                    channel := pid(),
-                   host := binary(),
+                   host_type := mongooseim:host_type_or_global(),
                    pool_tag := atom(),
                    confirms := boolean(),
                    max_queue_len := non_neg_integer() | infinity}.
 
 -type worker_opts() :: state().
+-type publish_result() :: boolean() | timeout | {channel_exception, any(), any()}.
 
 %%%===================================================================
 %%% Metrics
 %%%===================================================================
 
--define(CONNECTIONS_ACTIVE_METRIC(Tag), [backends, Tag, connections_active]).
--define(CONNECTIONS_OPENED_METRIC(Tag), [backends, Tag, connections_opened]).
--define(CONNECTIONS_CLOSED_METRIC(Tag), [backends, Tag, connections_closed]).
--define(CONNECTIONS_FAILED_METRIC(Tag), [backends, Tag, connections_failed]).
--define(MESSAGES_PUBLISHED_METRIC(Tag), [backends, Tag, messages_published]).
--define(MESSAGES_FAILED_METRIC(Tag), [backends, Tag, messages_failed]).
--define(MESSAGES_TIMEOUT_METRIC(Tag), [backends, Tag, messages_timeout]).
--define(MESSAGE_PUBLISH_TIME_METRIC(Tag),
-        [backends, Tag, message_publish_time]).
--define(MESSAGE_PAYLOAD_SIZE_METRIC(Tag),
-        [backends, Tag, message_payload_size]).
+-define(CONNECTIONS, mongoose_wpool_rabbit_connections).
+-define(MESSAGES_PUBLISHED, mongoose_wpool_rabbit_messages_published).
```

Comment:
I used `mongoose_wpool` as a prefix, because these events are for `mongoose_wpool_rabbit`'s sake and for example for modules wejust used the module names as the event prefix. Would just `wpool_rabbit_` as a prefix be better? I'd think then it'd be better to change `mongoose_wpool_rdbms` events similarly.

Good note about the macros - now that I think about it I agree - I will get rid of them.

Reply:
I'd just skip all `mongoose` prefixes. Of course we could do it later, but having simple prefixes like `auth_`, `sm_`, `router_`, etc. seems more consistent. Also, when browsing metrics ordered alphabetically, you don't get some of them bunched up at `mongoose_`.